src/Container.php

```diff
@@ -164,6 +171,50 @@ protected function processDefinition($definition): void
         }
     }
 
+    private function getCacheCondition(array $definition): array
+    {
+        return $definition['__cacheCondition'];
+    }
+
+    private function hasCacheCondition($definition): bool
+    {
+        if (is_array($definition) && isset($definition['__cacheCondition']) && is_array($definition['__cacheCondition'])) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private function clearDefinitionCacheCondition(array $definition)
+    {
+        unset($definition['__cacheCondition']);
+        if (isset($definition['__definition'])) {
+            $definition = $definition['__definition'];
+        }
+
+        return $definition;
+    }
+
+    private function validateCache($id): void
+    {
+        if (isset($this->cacheConditions[$id])) {
+            foreach ($this->cacheConditions[$id] as $service => $dependency) {
+                $currentValue = $service;
+                $dependencyValue = $dependency;
+                if (is_callable($dependency)) {
+                    $dependencyValue = $dependency($this);
+                }
+                if ($this->has($service)) {
+                    $currentValue = $this->get($service);
+                }
+                if ($currentValue != $dependencyValue) {
```

Comment:
```suggestion
                if ($currentValue !== $dependencyValue) {
```
It's simple to use equal values on the left and on the right side. On the other hand there are lots of errors based on type conversion.

Reply:
It makes sense. But this is just a draft for now.

Reply:
An array key can't be boolean, it is converted to an integer value. So, if we want to use `strong equal` and the boolean key, the right part must returns `1` or `0` only.

Reply:
Done